packages/pangea-sdk/PangeaCyber.Net/Utils/Crypto.cs

```diff
@@ -62,5 +79,151 @@ public static string AsymmetricPemExport(AsymmetricKeyParameter publicKey)
+        ///
+        /// </summary>
+        /// <param name="cipherText"></param>
+        /// <param name="encryptedSalt"></param>
+        /// <param name="iv"></param>
+        /// <param name="privateKey"></param>
+        /// <param name="asymmetricAlgorithm"></param>
+        /// <param name="symmetricAlgorithm"></param>
+        /// <param name="password"></param>
+        /// <param name="hashAlgorithm"></param>
+        /// <param name="iterationCount"></param>
+        /// <param name="kdf"></param>
+        /// <returns></returns>
+        public static byte[] KEMDecrypt(
+            byte[] cipherText,
+            byte[] encryptedSalt,
+            byte[] iv,
+            AsymmetricKeyParameter privateKey,
+            string asymmetricAlgorithm,
+            string symmetricAlgorithm,
+            string password,
+            string hashAlgorithm,
+            int iterationCount,
+            string kdf
+        )
+        {
+
+            // FIXME: compare against enum value
+            if (symmetricAlgorithm != "AES-GCM-256")
+            {
+                throw new Exception(string.Format("invalid symmetricAlgorithm: {0}", symmetricAlgorithm));
+            }
+
+            // FIXME: compare against enum value
+            if (asymmetricAlgorithm != "RSA-NO-PADDING-4096-KEM")
+            {
+                throw new Exception(string.Format("invalid asymmetricAlgorithm: {0}", symmetricAlgorithm));
+            }
+
+            if (kdf != "pbkdf2")
+            {
+                throw new Exception(string.Format("invalid kdf: {0}", kdf));
+            }
+
+            if (hashAlgorithm != "sha512")
+            {
+                throw new Exception(string.Format("invalid hashAlgorithm: {0}", hashAlgorithm));
+            }
+            var hashAlgorithmName = KeyDerivationPrf.HMACSHA512;
+
+            // Decrypt no padding
+            byte[] salt = RSADecryptNoPadding(encryptedSalt, privateKey);
+            var keyLength = GetKeyLength(symmetricAlgorithm);
+            var symmetricKey = KeyDerivation.Pbkdf2(password, salt, hashAlgorithmName, iterationCount, keyLength);
+
+            var plainText = AESGCMDecrypt(symmetricKey, cipherText, iv, null);
+            return plainText;
+        }
+
+        private static byte[] RSADecryptNoPadding(byte[] cipherText, AsymmetricKeyParameter privateKey)
+        {
+            var rsaEngine = new RsaEngine();
+            rsaEngine.Init(false, privateKey); // false = decryption mode
+            return rsaEngine.ProcessBlock(cipherText, 0, cipherText.Length);
+        }
+
+        /// <summary>
+        /// AES GCM Decryption
+        /// </summary>
+        /// <param name="key"></param>
+        /// <param name="cipherText"></param>
+        /// <param name="nonce"></param>
+        /// <param name="associatedData"></param>
+        /// <returns></returns>
+        public static byte[] AESGCMDecrypt(byte[] key, byte[] cipherText, byte[] nonce, byte[]? associatedData)
```

Comment:
Maybe keep this private? Just trying to keep the API surface small, and I believe the expectation is that users will use `KEMDecrypt()` instead.

Reply:
Fixed!